workshops/views.py

```diff
@@ -1194,7 +1214,20 @@ def event_edit(request, event_ident):
 
             else:
                 messages.error(request, 'Fix errors below.')
+        if 'sponsor-event' in request.POST:
+            sponsor_form = SponsorshipForm(request.POST, prefix='sponsor')
 
+            if sponsor_form.is_valid():
+                sponsor = sponsor_form.save()
+
+                messages.success(
+                    request,
+                    '{} was added a new sponsor.'.format(sponsor.organization)
+                )
+                # to reset the form values
+                return redirect(request.path)
```

Comment:
…or `redirect(event)`, right? The latter is less popular but IMO a little bit clearer where you end up after the redirect.